### PR TITLE
feat: add figma code connect

### LINKED ISSUE #16561: Figma Code Connect: Set up CI / github actions to automatically publish updates

### PR DESCRIPTION
**Pull request by alisonjoseph**
Ref https://github.com/carbon-design-system/carbon/issues/16273
Closes #16561

#### Changelog

**New**

- add `@figma/code-connect`
- add github action to publish code connect whenever a `*.figma.tsx` file is added/updated
- add `FIGMA_ACCESS_TOKEN` to secrets
- add `figma.config.json` config
- add `code-connect/README.md` file 
- exclude `*.figma.tsx` files from tsconfig
- Connect components
  - Accordion
  - Breadcrumb
  - Button
  - Checkbox
  - CodeSnippet
  - ContainedList
  - ContentSwitcher
  - Dropdown
  - Link
  - Notification
  - NumberInput
  - Tabs
  - Toggle

*will add additional components in separate PRs

### CODE REVIEW

File: .github/workflows/code-connect.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+      - src/packages/react/code-connect/**/*.figma.tsx
+    branches:
+      - main
+
+jobs:
+  code-connect:
+    name: Code Connect
+    runs-on: ubuntu-latest
+    steps:
+      - run: npx figma connect publish
```
Comment (tay1orjones) [inline .github/workflows/code-connect.yml:13]: I see `FIGMA_ACCESS_TOKEN` was added to the repo action secrets 👍

**Review by tay1orjones — APPROVED** [review]
🔥 🚢 🚀

**Review by guidari — APPROVED** [review]

**Comment by dlavrenuek** [comment]
@alisonjoseph is it correct that `@figma/code-connect` is listed as a production dependency instead of dev-dependency? The described functionality does not seem to be added to production code when `@carbon/react` is included in a project.

**Comment by alisonjoseph** [comment]
@dlavrenuek you're correct, updated here https://github.com/carbon-design-system/carbon/pull/16834

**Pull request merged**